### PR TITLE
Feature/listing api handler

### PR DESCRIPTION
**Pull request by oldben87**
## Why

Create a page to use the listings api endpoint.
Created an item select box that can return the associated item id and name onSelectChange or onClick

## Related Tickets
#81 
#79 

## Considerations

Might be worth looking at the listings icon button, might be worth coming up with some better names so they are clearer what we are going to be getting.

Additional context/info.

### CODE REVIEW

File: app/requests/GetListing.ts [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+}
+
+const GetListing: ({
+  itemId,
+  world,
+  daysRange
+}: GetListingProps) => Promise<Response> = async ({
+  itemId,
+  world,
+  daysRange = [7, 0]
```
Comment (cohenaj194) [inline app/requests/GetListing.ts:43]: This shouldn't be hard coded. We need to let users pick these numbers add text input so they can do this.

File: app/routes/queries/listings/Results.tsx [hunk 2]
```diff
@@ -0,0 +1,194 @@
...
+  return itemRank.passed
+}
+
+const Results = ({ data }: { data: ListingResponseType }) => {
+  const [columnFilters, setColumnFilters] = useState<ColumnFiltersState>([])
+  const [globalFilter, setGlobalFilter] = useState('')
+  const [columnOrder] = useState<ColumnOrderState>([])
+
+  const columnHelper = createColumnHelper<Listing>()
+  const columns = [
```
Comment (Acen) [inline app/routes/queries/listings/Results.tsx:46]: Jeeze I need to abstract this away. Well done managing the copy pasta :X

**Review by Acen — COMMENTED** [review]
Nice work!
This makes me want to abstract all the detail I've forced people to append >_<

File: app/routes/queries/listings/Results.tsx (hunk 2 above)
Comment (ButternCream) [reply app/routes/queries/listings/Results.tsx:46]: lol git good

Comment (Acen) [reply app/routes/queries/listings/Results.tsx:46]: i cri err tim

**Comment by cohenaj194** [comment]
Some notes:
- display the average and median price diff and time diff.
- time diff is in minutes not hours
- the background of those boxes can change with the results
- if a price diff is over 10000 make it red, lower make it green
- if a time diff is under 30 min make it red, lower make it green

File: app/components/navigation/sidebar/Sidebar.tsx [hunk 3]
```diff
@@ -33,6 +33,11 @@ const navigation = [
     href: 'queries',
     icon: ChartSquareBarIcon
   },
+  {
+    name: 'Lisings',
```
Comment (Acen) [inline app/components/navigation/sidebar/Sidebar.tsx:37]: Such typo, much wow

**Review by Acen — CHANGES_REQUESTED** [review]

File: app/components/navigation/sidebar/Sidebar.tsx (hunk 3 above)
Comment (cohenaj194) [reply app/components/navigation/sidebar/Sidebar.tsx:37]: lol that was me, but yea that should. be fixed

**Comment by cohenaj194** [comment]
Actually ignore my last comment i need to test more, my updated notes.

- display the average and median price diff and time diff values
- time diff is in minutes not hours (colors are correct)
- if a price diff is over 10000 make it red, lower make it green

**Comment by Acen** [comment]
> Actually ignore my last comment i need to test more, my updated notes.
> 
> * display the average and median price diff and time diff values
> * time diff is in minutes not hours (colors are correct)
> * if a price diff is over 10000 make it red, lower make it green

Protip: choose the red/green colours.
Here's a link to all the currently available ones:
https://tailwindcss.com/docs/customizing-colors

**Comment by cohenaj194** [comment]
Having this bug
https://user-images.githubusercontent.com/17516896/195223630-6ea5f006-fe75-416e-92bf-475feaaa3409.mov

**Comment by cohenaj194** [comment]
2 bugs to fix: 

* swap the colors on price diff so that under 10000 is green and over is red
* time diff is in minutes not hours

**Comment by cohenaj194** [comment]
So when there are no listings for an item and the api response is `{}` this is the output 

image

The error message should instead say something like `The item you searched for is out of stock and has no listings on your home server.`

**Pull request merged**